R/likelihood.R

```diff
@@ -110,8 +121,20 @@ likelihood <- function(chains, statistic = c("size", "length"), offspring_dist,
     likelihoods[sx[!(sx %in% exclude)]]
   })
 
-  if (!individual) {
-    chains_likelihood <- vapply(chains_likelihood, sum, 0)
+  ## transform log-likelihoods into likelihoods if required
+  if (!isTRUE(log)) {
+    chains_likelihood <- lapply(chains_likelihood, exp)
+  }
+
+  ## if individual == FALSE, return the joint log-likelihood
+  ## (sum of the log-likelihoods), if log == TRUE, else
+  ## multiply the likelihoods
+  if (!isTRUE(individual)) {
+    if (isTRUE(log)) {
+      chains_likelihood <- vapply(chains_likelihood, sum, 0)
+    } else {
+      chains_likelihood <- vapply(chains_likelihood, prod, 0)
+    }
```

Comment:
For numerical stability it would probably be better to do the sum on the log-likelihood first and then exponentiate - or, if we rename the function to `log-likelihood` and drop the `log` argument as I suggest above, we can just do the sum.

Reply:
I've converted this to an issue especially because of the latter consideration.